vignettes/stata.Rmd

```diff
@@ -0,0 +1,190 @@
+
+Average marginal effects                        Number of obs     =         32
+Model VCE    : OLS
+ 
+Expression   : Linear prediction, predict()
+dy/dx w.r.t. : cyl hp wt
+ 
+------------------------------------------------------------------------------
+    |            Delta-method
+    |      dy/dx   Std. Err.      t    P>|t|     [95% Conf. Interval]
+-------------+----------------------------------------------------------------
+cyl |  -.9416168   .5509164    -1.71   0.098    -2.070118    .1868842
+ hp |  -.0180381   .0118762    -1.52   0.140    -.0423655    .0062893
+ wt |  -3.166973   .7405759    -4.28   0.000    -4.683974   -1.649972
+------------------------------------------------------------------------------
+```
+
+### marginaleffects
+
+The same results can be obtained with `marginaleffects()` and `summary()` like this:
+
+```{r}
+library("marginaleffects")
+mod <- lm(mpg ~ cyl + hp + wt, data = mtcars)
+mfx <- marginaleffects(mod)
+summary(mfx)
+```
+
+
+## Counterfactual Marginal Effects
+
+### Stata
+
+With the `at` argument, Stata's `margins` command estimates average _counterfactual_ marginal effects. Here is an example:
+
+```
+quietly reg mpg i.cyl##c.hp wt
+margins, dydx(hp) at(cyl = (4 6 8))
+
+Average marginal effects                        Number of obs     =         32
+Model VCE    : OLS
+
+Expression   : Linear prediction, predict()
+dy/dx w.r.t. : hp
+
+1._at        : cyl             =           4
+
+2._at        : cyl             =           6
+
+3._at        : cyl             =           8
+
+------------------------------------------------------------------------------
+             |            Delta-method
+             |      dy/dx   Std. Err.      t    P>|t|     [95% Conf. Interval]
+-------------+----------------------------------------------------------------
+hp           |
+         _at |
+          1  |   -.099466   .0348665    -2.85   0.009    -.1712749   -.0276571
+          2  |  -.0213768    .038822    -0.55   0.587    -.1013323    .0585787
+          3  |   -.013441   .0125138    -1.07   0.293    -.0392137    .0123317
+------------------------------------------------------------------------------
+```
+
+<!-- Adapted from this GitHub issue: https://github.com/strengejacke/ggeffects/issues/249 -->
+
+Concretely, this is what Stata does in the background:
+
+  1. It duplicates the whole dataset 3 times and sets the values of `cyl` to the three specified values in each of those subsets.
+  2. It calculates marginal effects for that large grid.
+  3. It takes the average marginal effect for each value of cyl. 
+
+### marginaleffects
+
+You can estimate average counterfactual marginal effects with `marginaleffects()` by, first, setting the `grid.type` argument of `data.grid()` to `"counterfactual"` and, second, by averaging these effects, for example using `aggregate()` or `dplyr::summarise()`. 
+
+```{r}
+mod <- lm(mpg ~ as.factor(cyl) * hp + wt, data = mtcars)
+
+nd <- datagrid(model = mod,
```

Comment:
See comment below. I think it's more convenient to use `datagrid` in the main function call with the `newdata` argument.